modules/FaceExpressions_module/src/FaceExpressionsRetargeting.cpp

```diff
@@ -0,0 +1,181 @@
+         * rate goes up.
+         *
+         * Valid modes are 0 ("quality"), 1 ("low bitrate"), 2 ("aggressive"), and 3
+         * ("very aggressive"). The default mode is 0.
+         *
+         * Returns 0 on success, or -1 if the specified mode is invalid.
+         */
+    fvad_set_mode(m_fvadObject, 3); //Using 3 as the other methods are indeed too sensitive for our application.
+
+    /*
+         * Sets the input sample rate in Hz for a VAD instance.
+         *
+         * Valid values are 8000, 16000, 32000 and 48000. The default is 8000. Note
+         * that internally all processing will be done 8000 Hz; input data in higher
+         * sample rates will just be downsampled first.
+         *
+         * Returns 0 on success, or -1 if the passed value is invalid.
+         */
+    if (fvad_set_sample_rate(m_fvadObject, VAD_FREQUENCY))
+    {
+        yError() << "Unsupported input frequency.";
+        return false;
+    }
+
+    m_copiedSound.resize(VAD_SAMPLE_LENGTH);
+
+    yInfo() << "Started";
+
+
+    return true;
+}
+
+double FaceExpressionsRetargetingModule::getPeriod()
+{
+    std::lock_guard<std::mutex> lock(m_mutex);
+
+    return m_period;
+}
+
+bool FaceExpressionsRetargetingModule::updateModule()
+{
+    std::lock_guard<std::mutex> lock(m_mutex);
+
+    if (m_audioPort.getPendingReads())
+    {
+        yarp::sig::Sound* inputSound = m_audioPort.read(false);
+
+        if (inputSound->getFrequency() < VAD_FREQUENCY)
+        {
+            yError() << "The frequency needs to be at least " << VAD_FREQUENCY;
+            return false;
+        }
+
+        int subsampling = std::round(inputSound->getFrequency() / (double)VAD_FREQUENCY);
+
+        size_t soundLength = VAD_SAMPLE_LENGTH * subsampling;
+
+        if (soundLength > inputSound->getSamples())
+        {
+            yError() << "The input sound is too short.";
+            return false;
+        }
+
+        for (size_t i = 0; i < VAD_SAMPLE_LENGTH; ++i)
+        {
+            m_copiedSound[i] = inputSound->get(i*subsampling);
+        }
+
+        /*
+         * Calculates a VAD decision for an audio frame.
+         *
+         * `frame` is an array of `length` signed 16-bit samples. Only frames with a
+         * length of 10, 20 or 30 ms are supported, so for example at 8 kHz, `length`
+         * must be either 80, 160 or 240.
+         *
+         * Returns              : 1 - (active voice),
+         *                        0 - (non-active Voice),
+         *                       -1 - (invalid frame length).
+         */
+        m_isTalking = fvad_process(m_fvadObject, m_copiedSound.data(), m_copiedSound.size());
+
+        if (m_isTalking < 0)
+        {
+            yError() << "Invalid frame length.";
+            return false;
+        }
+
+        if (m_isTalking)
+        {
+            m_switchCounter++;
+
+            if (m_switchCounter > m_switchValue)
```

Comment:
So, here considering the values of `VAD_FREQUENCY` and  `VAD_SAMPLE_LENGTH`, while talking, every `20ms` we change the mouse values between `sur` and `nue`. Am I right @S-Dafarra ?